### ISSUE TITLE
JS typings, missing file reference for error TS2355 when tyiping functions with @type

### ISSUE THREAD

**Issue by dbellavista** [microsoft/TypeScript#43160]
# Bug Report

In javascript when defining a callback and applying the type to a function which doesn't return anything, an error of type TS2355 is raised. And this is the correct behavior.

However `tsc` fails to show both the file and the line number of the problem. This can be quite annoying when dealing with a very big project :D

### 🔎 Search Terms

* TS2355: A function whose declared type is neither 'void' nor 'any' must return a value.
* jsdoc @type @callback

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about javascript

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?useJavaScript=true#code/PQKhCgAIUgBAzAlgGwKaUQZ0wV1QOgCtMoZYBbAewBMc0Ns9gtdVS4BDHAFwAtKATpAAiHAHaJU9AEJTkHAG5ZuHUsHAAiHJnSZuAxAGNuGgNzhwoCNDiGOyZACMOhgNaQAYjjGH2sAajcOAJimJAA3gAKApTkWKgAPGI45I6oAgB8AL5qFlZw3ACeAA7o4V4+WdDq8N7GiJRikNyoegAUAJQRUJCGjZiUaPjIlADmbRoAcpSlGh3mOZZgBSVlFYZVIOocmIU+kLU+3A1NLXoATJ3dkJABQSGQGjvUZuBZQA)

### 💻 Code

```javascript
/**
 * @callback Func
 * @returns {Promise<number>}
 */

/** @type {Func} */
function test() {
  console.log("Nope");
}
```

### 🙁 Actual behavior

- The error TS2355 is correctly detected but the line number (and also the file when dealing with a project) is not reported

### 🙂 Expected behavior

- The tsc compiler should correctly report both line number and file
